crypto/src/hash.rs

```diff
@@ -19,28 +16,20 @@ use crate::{
 ///
 /// We require that the input be castable to a vector of u64s; this is to
 /// match the hashes defined in the ZK circuitry
-pub fn poseidon_hash_default_params<T: Into<Vec<u64>>>(val: T) -> DalekRistrettoField {
-    let hashable_values: Vec<u64> = val.into();
-    let arkworks_input = hashable_values
-        .into_iter()
-        .map(DalekRistrettoField::from)
-        .collect_vec();
-
+pub fn poseidon_hash_default_params<T: Into<Scalar>>(val: Vec<T>) -> Scalar {
+    let input = val.into_iter().map(|x| x.into().inner()).collect_vec();
```

Comment:
not able to find an `inner` method on the `Scalar` type either - is the branch of `mpc-stark` you're using here pushed up to github?

Reply:
correct, it won't be pushed yet, it'll get merged in as a patch version bump. Not worth creating an analogous PR for every small method I need to add